dev/cuda/matmul_forward.cu

```diff
@@ -72,11 +72,9 @@ __global__ void matmul_forward_kernel1(float* out,
     int bt = blockIdx.x * blockDim.x + threadIdx.x;
     int oc = blockIdx.y * blockDim.y + threadIdx.y;
     if (bt < BT && oc < OC) {
-        int b = bt / BT;
-        int t = bt % BT;
         float val = (bias != NULL) ? bias[oc] : 0.0f;
-        const float* wrow = weight + oc*C;
-        const float* inp_bt = inp + b * BT * C + t * C;
+        const float* wrow = weight + oc * C;
+        const float* inp_bt = inp + bt * C;
```

Comment:
since the inp is [B x T x C] and you set the kernel grid as gridDim(ceil_div(B * T, sqrt_block_size), ceil_div(OC, sqrt_block_size)) meaning that bt = blockIdx.x * blockDim.x + threadIdx.x itsself an index into BT axis. So when you visit inp, the expression inp_bt = inp + bt * C is self explainary. 

when i first read the origin code i got confused why b and t are used here given b is always zero for sure even at the cost of expensive / and % operation in CUDA threads.

Reply:
Thanks for your reponse Azret